specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json

```diff
@@ -0,0 +1,9155 @@
+            "AzureSqlDatabaseManagedInstance",
+            "AzureSqlDataWarehouse",
+            "AzureMySql",
+            "AzureStorage",
+            "Teradata",
+            "Oracle",
+            "SapS4Hana",
+            "SapEcc",
+            "PowerBI"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "DataSourceType",
+            "modelAsString": true
+          }
+        },
+        "scans": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Scan"
+          },
+          "readOnly": true
+        }
+      },
+      "discriminator": "kind"
+    },
+    "DataSourceList": {
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DataSource"
+          }
+        },
+        "nextLink": {
+          "type": "string"
+        },
+        "count": {
+          "format": "int64",
+          "type": "integer"
+        }
+      }
+    },
+    "FilterProperties": {
+      "required": [
+        "excludeUriPrefixes",
+        "includeUriPrefixes"
+      ],
+      "type": "object",
+      "properties": {
+        "excludeUriPrefixes": {
+          "uniqueItems": true,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "includeUriPrefixes": {
+          "uniqueItems": true,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "excludeRegexes": {
+          "uniqueItems": true,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "includeRegexes": {
+          "uniqueItems": true,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "Filter": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/FilterProperties"
+            }
+          ]
+        }
+      }
+    },
+    "IntegrationRuntime": {
+      "required": [
+        "kind"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "kind": {
+          "enum": [
+            "SelfHosted",
+            "Managed"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "IntegrationRuntimeType",
+            "modelAsString": true
+          }
+        }
+      },
+      "discriminator": "kind"
+    },
+    "IActionResult": {
+      "type": "object"
+    },
+    "IntegrationRuntimeStatus": {
+      "required": [
+        "kind"
+      ],
+      "type": "object",
```

Comment:
these polymorphism without any meaningful shared property is painful.

Reply:
The `IntegrationRuntime` is added with your PR. I don't see why improving it would disturb any other external models that not in your PR.

That said, I am fine with it as there is already too many such polymorphism in scanning.

Reply:
It wouldn't disturb other models, but relying on `Kind` is used at a BaseAdapter level and we have to write some other adapter and conversion flow from ground up for this alone. Since there is a root in the source code where this change can start from and spread across entire repo, I felt this can be postponed until then.